domain/modelconfig/validators/validators.go

```diff
@@ -30,6 +30,37 @@ func CharmhubURLChange() config.ValidatorFunc {
 	}
 }
 
+// AgentStreamChange returns a config validator that will check to make sure the
+// agent stream does not change and also remove it from config so that it does
+// not get committed back to state.
+//
+// Agent stream is an ongoing value that is being actively removed from model
+// config. Until we can finish removing all uses of agent version from model
+// config this validator will keep removing it from the new config so that it
+// does not get persisted to state.
```

Comment:
Have we checked that on model migration the value is stripped when the model is imported

Reply:
No I am working on this story now in the next PR. Next PR will explicitly control the value and the default at bootstrap, model creation and migration.